src/com/sittinglittleduck/DirBuster/Worker.java

```diff
@@ -573,20 +297,171 @@ else if(work.getBaseCaseObj().isUseRegexInstead())
+            {
+                Thread.sleep(100);
+            }
+        }
+        /*
+         * Send the request
+         */
+        int code = httpclient.executeMethod(httpMethod);
+        
+        if(Config.debug)
+        {
+            System.out.println("DEBUG Worker[" + threadId + "]: " + code + " " + url.toString());
+        }
+		return code;
     }
 
+	private boolean verifyIfCodeIsValid(int code) {
+		return code != HttpStatus.SC_NOT_FOUND && code != 0 && code != HttpStatus.SC_BAD_GATEWAY;
+	}
+
+
+	private void verifyResponseForValidRequests(int code, String response, String rawResponse) {
+		if(Config.debug)
+		{
+		    System.out.println("DEBUG Worker[" + threadId + "]: Base Case Check " + url.toString());
+		}
+
+
+		//TODO move this option to the Adv options
+		//if the response does not match the base case
+		Pattern regexFindFile = Pattern.compile(".*file not found.*", Pattern.CASE_INSENSITIVE);
+
+		Matcher m = regexFindFile.matcher(response);
+
+		//need to clean the base case of the item we are looking for
+		String basecase = FilterResponce.removeItemCheckedFor(work.getBaseCaseObj().getBaseCase(), work.getItemToCheck());
+
+		if(m.find())
+		{
+			System.out.println("DEBUG Worker[" + threadId + "]: 404 for: " + url.toString());
+		}
+		else if(!response.equalsIgnoreCase(basecase))
+		{
+			notifyItemFound(code, response, rawResponse, basecase);
+		}
+	}
+	
+	private void notifyItemFound(int code, String response, String rawResponse, String basecase, String type) {
+		if(work.isDir())
+		{
+		    if(Config.debug)
+		    {
+		        System.out.println("DEBUG Worker[" + threadId + "]: Found Dir (" + type +")" + url.toString());
+		    }
+		    //we found a dir
+		    manager.foundDir(url, code, response, basecase, rawResponse, work.getBaseCaseObj());
+		}
+		else
+		{
+		    //found a file
+		    if(Config.debug)
+		    {
+		        System.out.println("DEBUG Worker[" + threadId + "]: Found File (" + type +")" + url.toString());
+		    }
+		    manager.foundFile(url, code, response, basecase, rawResponse, work.getBaseCaseObj());
+		}
+	}
+	
+
+	private void notifyItemFound(int code, String response, String rawResponse, String basecase) {
+		notifyItemFound(code, response, rawResponse, basecase, "base case");		
+	}
+
+	private void parseHtml(HttpMethodBase httpMethod, String response) {
+		//parse the html of what we have found
+
+	    Header contentType = httpMethod.getResponseHeader("Content-Type");
+
+	    if(contentType != null)
+	    {
+	        if(contentType.getValue().startsWith("text"))
+	        {
+	            manager.addHTMLToParseQueue(new HTMLparseWorkUnit(response, work));
+	        }
+	    }
+		
+	}
+
+	private String getResponseAsString(HttpMethodBase httpMethod) throws IOException {
+		StringBuilder builder = new StringBuilder();
+		
+		try(BufferedReader input = new BufferedReader(new InputStreamReader(httpMethod.getResponseBodyAsStream()))) {
+			builder.append(System.lineSeparator());
```

Comment:
This will lead to different content if the response's newline chars are not the same as the system's line separator.
It's better to convert to `String` from bytes than use the `Reader`, that way we don't need to care which newline characters are being used, for example:
`new String(httpMethod.getResponseBody(), Charset.forName(charset))` (with `charset` being defined in `Content-Type`, defaulting to UTF-8).